### PR TITLE
Fix loading config from database

### PR DESCRIPTION
**Pull request by nzpr**
## Overview

Loading of config from database was not operational and using custom json codecs.

This PR uses circe library to encode/decode config values.
Another change is removing nesting of config classes since it makes it harder to address config values using keys.

### Please make sure that this PR:

- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).

### CODE REVIEW

File: db/src/main/scala/slick/Actions.scala [hunk 1]
```diff
@@ -21,11 +20,11 @@ final case class Actions(profile: JdbcProfile, ec: ExecutionContext) {
 
   /** Config */
 
-  def putConfig(key: String, value: Any): DBIOAction[Int, NoStream, Write] =
-    qConfigs.insertOrUpdate((key, JsonCodec.toJsonString(value)))
```
Comment (nzpr) [inline db/src/main/scala/slick/Actions.scala:25]: If there is a need to encode something into JSON it should be done outside of a database logic.

File: db/src/test/scala/slick/SlickSpec.scala [hunk 2]
```diff
@@ -196,36 +196,6 @@ class SlickSpec extends AsyncFlatSpec with Matchers with ScalaCheckPropertyCheck
       .unsafeRunSync()
   }
 
-  "Configs saved to DB and loaded from DB" should "be the same" in {
```
Comment (nzpr) [inline db/src/test/scala/slick/SlickSpec.scala:199]: I think no need for this test, config storage is just reading/writing encoded values. Values are of a special type and codecs are circe. So type safety and having a single codec should do the job.

File: node/src/main/scala/node/Config.scala [hunk 3]
```diff
@@ -19,78 +22,14 @@ final case class Config(
   enableInfluxDb: Boolean = false,
   @Description("Enable dev mode. WARNING: This mode is not secure and should not be used in production.")
   devMode: Boolean = false,
-  @Description("WebApi configuration")
-  webApi: node.api.web.Config = node.api.web.Config("localhost", 8080),
```
Comment (nzpr) [inline node/src/main/scala/node/Config.scala:23]: I flattened configuration, if there are nested configs it is harder to maintain key paths and just encode/decode config objects.

File: db/src/test/scala/slick/SlickSpec.scala (hunk 2 above)
Comment (stanislavlyalin) [reply db/src/test/scala/slick/SlickSpec.scala:199]: This test also tests reflection, so I think it makes sense.

File: sdk/src/main/scala/sdk/reflect/ClassesAsConfig.scala [hunk 4]
```diff
@@ -21,13 +21,15 @@ object ClassesAsConfig {
     }
     .mkString(s"\n")
 
+  def fields(x: Any): List[String] = ClassAsTuple(x).map { case (name, _, _) => name }.toList
+
   def kvMap(root: String, classes: Any*): Map[String, Any] = classes.flatMap { clz =>
     ClassAsTuple(clz).map { case (name, value, _) =>
       s"$root.${configName(clz)}.$name" -> value
     }
   }.toMap
```
Comment (stanislavlyalin) [inline sdk/src/main/scala/sdk/reflect/ClassesAsConfig.scala:30]: This method is no longer needed. It is only used in the `ReferenceConf` test, which can be rewritten like this
```scala
"this" should "produce configuration parameter names" in {
  val paramNames =
    (
      sim.Config.Default,
      node.Config.Default,
      diagnostics.metrics.Config.Default,
      db.Config.Default,
      node.comm.Config.Default,
    ).productIterator.toList
      .flatMap { cfg =>
        val configName = ClassesAsConfig.configName(cfg)
        ClassesAsConfig.fields(cfg).map(key => s"$configName.$key")
      }
      .mkString("\n")
  println(paramNames)
}
```

**Review by stanislavlyalin — APPROVED** [review]
LGTM

File: sdk/src/main/scala/sdk/reflect/ClassesAsConfig.scala (hunk 4 above)
Comment (nzpr) [reply sdk/src/main/scala/sdk/reflect/ClassesAsConfig.scala:30]: Nice trick with productIterator!

File: db/src/test/scala/slick/SlickSpec.scala (hunk 2 above)
Comment (nzpr) [reply db/src/test/scala/slick/SlickSpec.scala:199]: I've added round trip test https://github.com/nzpr/gorki-node/pull/26/commits/4590771d03794581712fec35e22c24410997381a

**Pull request merged**